mdapi/database/base.py

```diff
@@ -121,12 +118,10 @@ def main(self):
             # We have never downloaded this before...
             # so we have nothing to compare it against. Just return and say there
             # are "no differences".
-            servlogr.logrobjc.warning(
-                "[%s] Database empty - %s cannot compare" % (self.name, self.dbsB)
-            )
+            servlogr.logrobjc.warning(f"[{self.name}] Database empty - {self.dbsB} cannot compare")
             return set()
 
-        assert len(tablistA) == len(tablistB), "Cannot compare disparate databases"
+        assert len(tablistA) == len(tablistB), "Cannot compare disparate databases"  # noqa : S101
```

Comment:
Here’s documentation about why ruff flags this: https://docs.astral.sh/ruff/rules/assert/

In short, `assert` is a no-op if the Python interpreter runs with the `-O` or `-OO` flags. Here’s an alternative:

```suggestion
        if len(tablistA) != len(tablistB):
            raise ValueError("Cannot compare disparate databases")
```

Reply:
The likeliness of someone running tests in the production environment (i.e. using the `-0` and the `-00` as mentioned above) is small but admittedly, not zero. I wonder if we could let things be the way they are as `S101` does feel a bit nitpicky[1] to me.

Sauce
1. https://stackoverflow.com/a/68429294

Reply:
It would be nit-picky if the argument was only about the likelihood of the code being run optimized or not. In my opinion, this is not the case: asserts (in Python and other languages) express that a certain condition applies – if it doesn’t, it must be a bug in the program.

I’m not very familiar with mdapi code, but it looks to me as if it pulls a database from a remote repository, then attempts to recreate it from other metadata and compares the two? In this case, a simple syncing problem on the other end could cause issues here.

Reply:
If only I knew how to read, I would have gotten to know that you marked out the `assert` in a file named `mdapi/database/base.py` and not from a test file. The points I referenced still stand for a testing environments and I sense that we talked in parallels in the aforementioned comments - where I was going about tests and you were about the actual codebase. 

TLDR - Changing it now.

Reply:
Yeah, asserts in the test suite are fine!